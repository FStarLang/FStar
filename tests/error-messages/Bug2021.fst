module Bug2021

[@(expect_failure [66])]
let test0 () =
  let x = (fun r -> 2) in
  ()

[@(expect_failure [66])]
let test1 () =
  let x r = 2 in
  ()

[@(expect_failure [66])]
let test2 () =
  ();
  let blah #r : int =
    2
  in
  ()

(* Reports over `f` *)
[@(expect_failure [66])]
let test3 (f : (#x:int -> unit -> unit -> unit)) =
  let blah = f () () in
  ()

(* Reports over `f ()` *)
[@(expect_failure [66])]
let test4 (f : (unit -> #x:int -> unit -> unit)) =
  let blah = f () () in
  ()

(* Reports over `f`, since it raises that error first
 * and then typechecking is aborted. *)
[@(expect_failure [66])]
let test5 (f : (#i:int -> unit -> #x:int -> unit -> unit)) =
  let blah = f () () in
  ()
