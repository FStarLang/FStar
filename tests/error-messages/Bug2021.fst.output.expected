>> Got issues: [
* Error 66 at Bug2021.fst(5,15-5,16):
  - Failed to resolve implicit argument ?2
    of type Type
    introduced for user-provided implicit term at Bug2021.fst(5,15-5,16)

>>]
>> Got issues: [
* Error 66 at Bug2021.fst(10,8-10,9):
  - Failed to resolve implicit argument ?2
    of type Type
    introduced for user-provided implicit term at Bug2021.fst(10,8-10,9)

>>]
>> Got issues: [
* Error 66 at Bug2021.fst(16,12-16,13):
  - Failed to resolve implicit argument ?5
    of type Type
    introduced for user-provided implicit term at Bug2021.fst(16,12-16,13)

>>]
>> Got issues: [
* Error 66 at Bug2021.fst(24,13-24,14):
  - Failed to resolve implicit argument ?9
    of type Prims.int
    introduced for Instantiating implicit argument 'x'
  - See also Bug2021.fst(23,11-23,12)

>>]
>> Got issues: [
* Error 66 at Bug2021.fst(30,13-30,17):
  - Failed to resolve implicit argument ?10
    of type Prims.int
    introduced for Instantiating implicit argument 'x'
  - See also Bug2021.fst(29,11-30,17)

>>]
>> Got issues: [
* Error 66 at Bug2021.fst(36,13-36,17):
  - Failed to resolve implicit argument ?13
    of type Prims.int
    introduced for Instantiating implicit argument 'x'
  - See also Bug2021.fst(35,11-36,17)

>>]
